dsgrid/dataset/table_format_handler_base.py

```diff
@@ -109,6 +111,11 @@ def add_columns(
                 dataset_id=self.dataset_id,
             )
 
+        if "fraction" in df.columns:
```

Comment:
I am surprised to see this at the same level as `for column in column_models:`. Am I reading this code (including the changes in query/models.py) correctly to say that multiple supplemental dimensions could be in the column_model list, e.g., for multiple dimension types? Or does one call to add_columns always process a single supplemental dimension, such that column_models is a list of supplemental dimension records?

Reply:
`column_models` is essentially a list of dimension query names in one aggregation. There can be multiple per dimension type (`county, state`) and multiple dimensions. There could also be multiple supplemental dimensions of the same type (`state, census_region`).

Reply:
Okay--so one aggregation means that the fractions can be continually collected: fraction = 1.0 * fraction_1 * fraction_2 * ... and then all applied at the end. That seems reasonable to me (although I can't fully prove it to myself without spending a bunch more time).

Reply:
Yes. This is the same methodology used for mapping dataset dimensions to the project.